tests/tests/integration_tests.rs

```diff
@@ -359,6 +359,44 @@ async fn ws_server_should_stop_subscription_after_client_drop() {
 	assert!(matches!(close_err.close_reason(), &SubscriptionClosedReason::ConnectionReset));
 }
 
+#[tokio::test]
+async fn ws_server_cancel_stream_after_reset_conn() {
+	use futures::{channel::mpsc, SinkExt, StreamExt};
+	use jsonrpsee::{ws_server::WsServerBuilder, RpcModule};
+
+	let server = WsServerBuilder::default().build("127.0.0.1:0").await.unwrap();
+	let server_url = format!("ws://{}", server.local_addr().unwrap());
+
+	let (tx, mut rx) = mpsc::channel(1);
+	let mut module = RpcModule::new(tx);
+
+	module
+		.register_subscription("subscribe_never_produce", "n", "unsubscribe_never_produce", |_, sink, mut tx| {
+			// create stream that doesn't produce items.
+			let stream = futures::stream::empty::<usize>();
+			tokio::spawn(async move {
+				sink.add_stream(stream).await;
+				let send_back = Arc::make_mut(&mut tx);
+				send_back.feed(()).await.unwrap();
+			});
+			Ok(())
+		})
+		.unwrap();
+
+	server.start(module).unwrap();
+
+	let client = WsClientBuilder::default().build(&server_url).await.unwrap();
+	let _sub1: Subscription<usize> =
+		client.subscribe("subscribe_never_produce", None, "unsubscribe_never_produce").await.unwrap();
+	let _sub2: Subscription<usize> =
+		client.subscribe("subscribe_never_produce", None, "unsubscribe_never_produce").await.unwrap();
+
+	// terminate connection.
+	drop(client);
+	assert_eq!(Some(()), rx.next().await, "subscription stream should be terminated after the client was dropped");
+	assert_eq!(Some(()), rx.next().await, "subscription stream should be terminated after the client was dropped");
```

Comment:
I felt like I understood the test until I got here. I thought `rx` would produce `None` after the client was dropped and so it'd be `assert!(rx.next().await.is_none())`. What am I missing? :/

Reply:
haha, maybe it's more clear as you described it but the test actually sends a message on the channel when the subscription terminated.